docs/source/enterprise-guide/change-management.mdx

```diff
@@ -0,0 +1,537 @@
+change. We might choose to progressively release the change using feature flags
+or canary deploys.
+
+[dataloader]:
+  https://www.apollographql.com/docs/apollo-server/data/data-sources/#using-with-dataloader
+
+Because this change does not require changes to other components, it allows
+rolling back the subgraph service should an issue arise in the new code.
+
+### Progressive delivery techniques
+
+**Feature flags**: we might choose to leverage feature flags, breaking up the
+release into multiple steps, enabling a quick rollback if we discover a problem
+in the implementation. In order to "pin" users to a specific flag value and
+ensure a consistent experience during the rollout, pass a user-identifying
+header though each layer of the system.
+
+1. Author the resolver to provide both old and new codepaths gated by a feature
+   flag.
+2. Release the new code. The new codepath is enabled for zero users by default,
+   so the release should have no impact on current behavior.
+3. Progressively enable the feature flag for more users, monitoring error rates
+   or support inquiries.
+4. Once the feature flag is enabled for 100% of users, remove the old codepath
+   and perform a clean-up release.
+
+**Canary deploys**: instead of adding complexity to our resolver by supporting
+two codepaths and selecting one based on a feature flag, we can choose to simply
+author the new resolver and release it more carefully.
+
+1. Author the new resolver.
+2. Release the new code using a canary deployment mechanism.
+3. Observe error rates or other relevant metrics (either manually, or with an
+   automated progressive delivery controller like [Flagger][flagger]).
+4. If metrics are acceptable, release the new code to all nodes.
+5. If metrics are unacceptable, cancel the canary deployment, roll back the
+   release, and revert the code change.
+
+[flagger]: https://flagger.app/
+
+## Changes to gateway logic or configuration
+
+The gateway is critical infrastructure and should be treated as such. Although
+we can treat releases as standard rolling deploys, it's a very good idea to use
+progressive delivery techniques like canary or blue-green deploys.
+
+### Progressive delivery techniques
+
+Deploying gateway changes without schema changes using a blue-green deployment
+process is straightforward, whether we're manually deploying to separate node
+pools or clusters, or using a automated progressive delivery controller like
+[Flagger](flagger).
+
+If we're also using
+[out-of-band schema publishing](#release-alternative-out-of-band-schema-publishing)
+to make shared type or query planning changes, we might choose to have a third
+pool or cluster of gateway instances configured to use a separate variant. This
+allows a progressive delivery controller to automatically manage the blue-green
+rollout of two pools/clusters configured to use the same variant, while
+reserving the third pool/cluster for out-of-band changes.
```

Comment:
Same comment here re: references to blue-green deployment management at the gateway level in terms of schemas. Happy to chat more about your thoughts here, but I would remove these references without having a flushed out doc explaining how to do this at scale.